prow/github/fakegithub/fakegithub.go

```diff
@@ -151,11 +151,24 @@ func (f *FakeClient) GetRef(owner, repo, ref string) (string, error) {
 	return "abcde", nil
 }
 
-func (f *FakeClient) CreateStatus(owner, repo, ref string, s github.Status) error {
-	f.CreatedStatuses[ref] = append(f.CreatedStatuses[ref], s)
+func (f *FakeClient) CreateStatus(owner, repo, sha string, s github.Status) error {
+	if f.CreatedStatuses == nil {
+		f.CreatedStatuses = make(map[string][]github.Status)
+	}
+	statuses := f.CreatedStatuses[sha]
+	for i := range statuses {
+		if statuses[i].Context == s.Context {
+			statuses[i] = s
+		}
```

Comment:
This only ever updates pre-existing statuses. It never creates a new status. This might be more simple to represent as `map[string]map[string]github.Status` (sha -> context -> status)